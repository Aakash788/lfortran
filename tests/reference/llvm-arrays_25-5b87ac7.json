{
    "basename": "llvm-arrays_25-5b87ac7",
    "cmd": "lfortran --no-color --show-llvm {infile} -o {outfile}",
    "infile": "tests/../integration_tests/arrays_25.f90",
    "infile_hash": "62760681bc697f79bfd4cf5ec2e2103ca68d3c9a57d72dc74e379fea",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "llvm-arrays_25-5b87ac7.stdout",
    "stdout_hash": "920a8005693dd7f99b96e55a22c09c7f26eac973856ae2527d19ec25",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}