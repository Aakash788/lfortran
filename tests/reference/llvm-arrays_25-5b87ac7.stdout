; ModuleID = 'LFortran'
source_filename = "LFortran"

%model_t = type { %array* }
%array = type { i32*, i32, %dimension_descriptor*, i1, i32 }
%dimension_descriptor = type { i32, i32, i32 }

@0 = private unnamed_addr constant [2 x i8] c" \00", align 1
@1 = private unnamed_addr constant [2 x i8] c" \00", align 1
@2 = private unnamed_addr constant [5 x i8] c"%d%s\00", align 1
@3 = private unnamed_addr constant [2 x i8] c" \00", align 1
@4 = private unnamed_addr constant [2 x i8] c"\08\00", align 1
@5 = private unnamed_addr constant [3 x i8] c"%s\00", align 1
@6 = private unnamed_addr constant [2 x i8] c" \00", align 1
@7 = private unnamed_addr constant [2 x i8] c"\0A\00", align 1
@8 = private unnamed_addr constant [3 x i8] c"%s\00", align 1

define i32 @main(i32 %0, i8** %1) {
.entry:
  call void @_lpython_set_argv(i32 %0, i8** %1)
  %__1_k = alloca i32, align 4
  %__1_t = alloca i32, align 4
  %m = alloca %model_t, align 8
  %2 = getelementptr %model_t, %model_t* %m, i32 0, i32 0
  store %array* null, %array** %2, align 8
  %3 = getelementptr %model_t, %model_t* %m, i32 0, i32 0
  %4 = load %array*, %array** %3, align 8
  %5 = getelementptr %array, %array* %4, i32 0, i32 1
  store i32 0, i32* %5, align 4
  %6 = getelementptr %array, %array* %4, i32 0, i32 2
  %7 = load %dimension_descriptor*, %dimension_descriptor** %6, align 8
  %8 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %7, i32 0
  %9 = getelementptr %dimension_descriptor, %dimension_descriptor* %8, i32 0, i32 0
  %10 = getelementptr %dimension_descriptor, %dimension_descriptor* %8, i32 0, i32 1
  %11 = getelementptr %dimension_descriptor, %dimension_descriptor* %8, i32 0, i32 2
  store i32 1, i32* %9, align 4
  store i32 1, i32* %10, align 4
  store i32 10, i32* %11, align 4
  %12 = getelementptr %array, %array* %4, i32 0, i32 0
  %13 = alloca i32, align 4
  store i32 40, i32* %13, align 4
  %14 = load i32, i32* %13, align 4
  %15 = call i8* (i32, ...) @_lfortran_malloc(i32 %14)
  %16 = bitcast i8* %15 to i32*
  store i32* %16, i32** %12, align 8
  %17 = getelementptr %model_t, %model_t* %m, i32 0, i32 0
  %18 = load %array*, %array** %17, align 8
  %19 = getelementptr %array, %array* %18, i32 0, i32 2
  %20 = load %dimension_descriptor*, %dimension_descriptor** %19, align 8
  %21 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %20, i32 0
  %22 = getelementptr %dimension_descriptor, %dimension_descriptor* %21, i32 0, i32 1
  %23 = load i32, i32* %22, align 4
  %24 = sub i32 %23, 1
  store i32 %24, i32* %__1_t, align 4
  br label %loop.head

loop.head:                                        ; preds = %loop.body, %.entry
  %25 = load i32, i32* %__1_t, align 4
  %26 = add i32 %25, 1
  %27 = getelementptr %model_t, %model_t* %m, i32 0, i32 0
  %28 = load %array*, %array** %27, align 8
  %29 = getelementptr %array, %array* %28, i32 0, i32 2
  %30 = load %dimension_descriptor*, %dimension_descriptor** %29, align 8
  %31 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %30, i32 0
  %32 = getelementptr %dimension_descriptor, %dimension_descriptor* %31, i32 0, i32 1
  %33 = load i32, i32* %32, align 4
  %34 = getelementptr %dimension_descriptor, %dimension_descriptor* %31, i32 0, i32 2
  %35 = load i32, i32* %34, align 4
  %36 = add i32 %35, %33
  %37 = sub i32 %36, 1
  %38 = icmp sle i32 %26, %37
  br i1 %38, label %loop.body, label %loop.end

loop.body:                                        ; preds = %loop.head
  %39 = load i32, i32* %__1_t, align 4
  %40 = add i32 %39, 1
  store i32 %40, i32* %__1_t, align 4
  %41 = getelementptr %model_t, %model_t* %m, i32 0, i32 0
  %42 = load i32, i32* %__1_t, align 4
  %43 = load %array*, %array** %41, align 8
  %44 = getelementptr %array, %array* %43, i32 0, i32 2
  %45 = load %dimension_descriptor*, %dimension_descriptor** %44, align 8
  %46 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %45, i32 0
  %47 = getelementptr %dimension_descriptor, %dimension_descriptor* %46, i32 0, i32 1
  %48 = load i32, i32* %47, align 4
  %49 = sub i32 %42, %48
  %50 = getelementptr %dimension_descriptor, %dimension_descriptor* %46, i32 0, i32 0
  %51 = load i32, i32* %50, align 4
  %52 = mul i32 %51, %49
  %53 = add i32 0, %52
  %54 = getelementptr %array, %array* %43, i32 0, i32 1
  %55 = load i32, i32* %54, align 4
  %56 = add i32 %53, %55
  %57 = getelementptr %array, %array* %43, i32 0, i32 0
  %58 = load i32*, i32** %57, align 8
  %59 = getelementptr inbounds i32, i32* %58, i32 %56
  store i32 0, i32* %59, align 4
  br label %loop.head

loop.end:                                         ; preds = %loop.head
  %60 = getelementptr %model_t, %model_t* %m, i32 0, i32 0
  %61 = load %array*, %array** %60, align 8
  %62 = getelementptr %array, %array* %61, i32 0, i32 2
  %63 = load %dimension_descriptor*, %dimension_descriptor** %62, align 8
  %64 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %63, i32 0
  %65 = getelementptr %dimension_descriptor, %dimension_descriptor* %64, i32 0, i32 1
  %66 = load i32, i32* %65, align 4
  %67 = sub i32 5, %66
  %68 = getelementptr %dimension_descriptor, %dimension_descriptor* %64, i32 0, i32 0
  %69 = load i32, i32* %68, align 4
  %70 = mul i32 %69, %67
  %71 = add i32 0, %70
  %72 = getelementptr %array, %array* %61, i32 0, i32 1
  %73 = load i32, i32* %72, align 4
  %74 = add i32 %71, %73
  %75 = getelementptr %array, %array* %61, i32 0, i32 0
  %76 = load i32*, i32** %75, align 8
  %77 = getelementptr inbounds i32, i32* %76, i32 %74
  store i32 5, i32* %77, align 4
  %78 = getelementptr %model_t, %model_t* %m, i32 0, i32 0
  %79 = load %array*, %array** %78, align 8
  %80 = getelementptr %array, %array* %79, i32 0, i32 2
  %81 = load %dimension_descriptor*, %dimension_descriptor** %80, align 8
  %82 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %81, i32 0
  %83 = getelementptr %dimension_descriptor, %dimension_descriptor* %82, i32 0, i32 1
  %84 = load i32, i32* %83, align 4
  %85 = sub i32 %84, 1
  store i32 %85, i32* %__1_k, align 4
  br label %loop.head1

loop.head1:                                       ; preds = %loop.body2, %loop.end
  %86 = load i32, i32* %__1_k, align 4
  %87 = add i32 %86, 1
  %88 = getelementptr %model_t, %model_t* %m, i32 0, i32 0
  %89 = load %array*, %array** %88, align 8
  %90 = getelementptr %array, %array* %89, i32 0, i32 2
  %91 = load %dimension_descriptor*, %dimension_descriptor** %90, align 8
  %92 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %91, i32 0
  %93 = getelementptr %dimension_descriptor, %dimension_descriptor* %92, i32 0, i32 1
  %94 = load i32, i32* %93, align 4
  %95 = getelementptr %dimension_descriptor, %dimension_descriptor* %92, i32 0, i32 2
  %96 = load i32, i32* %95, align 4
  %97 = add i32 %96, %94
  %98 = sub i32 %97, 1
  %99 = icmp sle i32 %87, %98
  br i1 %99, label %loop.body2, label %loop.end3

loop.body2:                                       ; preds = %loop.head1
  %100 = load i32, i32* %__1_k, align 4
  %101 = add i32 %100, 1
  store i32 %101, i32* %__1_k, align 4
  %102 = getelementptr %model_t, %model_t* %m, i32 0, i32 0
  %103 = load i32, i32* %__1_k, align 4
  %104 = load %array*, %array** %102, align 8
  %105 = getelementptr %array, %array* %104, i32 0, i32 2
  %106 = load %dimension_descriptor*, %dimension_descriptor** %105, align 8
  %107 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %106, i32 0
  %108 = getelementptr %dimension_descriptor, %dimension_descriptor* %107, i32 0, i32 1
  %109 = load i32, i32* %108, align 4
  %110 = sub i32 %103, %109
  %111 = getelementptr %dimension_descriptor, %dimension_descriptor* %107, i32 0, i32 0
  %112 = load i32, i32* %111, align 4
  %113 = mul i32 %112, %110
  %114 = add i32 0, %113
  %115 = getelementptr %array, %array* %104, i32 0, i32 1
  %116 = load i32, i32* %115, align 4
  %117 = add i32 %114, %116
  %118 = getelementptr %array, %array* %104, i32 0, i32 0
  %119 = load i32*, i32** %118, align 8
  %120 = getelementptr inbounds i32, i32* %119, i32 %117
  %121 = load i32, i32* %120, align 4
  call void (i8*, ...) @_lfortran_printf(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @2, i32 0, i32 0), i32 %121, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @1, i32 0, i32 0))
  br label %loop.head1

loop.end3:                                        ; preds = %loop.head1
  call void (i8*, ...) @_lfortran_printf(i8* getelementptr inbounds ([3 x i8], [3 x i8]* @5, i32 0, i32 0), i8* getelementptr inbounds ([2 x i8], [2 x i8]* @4, i32 0, i32 0))
  call void (i8*, ...) @_lfortran_printf(i8* getelementptr inbounds ([3 x i8], [3 x i8]* @8, i32 0, i32 0), i8* getelementptr inbounds ([2 x i8], [2 x i8]* @7, i32 0, i32 0))
  ret i32 0
}

declare void @_lpython_set_argv(i32, i8**)

declare i8* @_lfortran_malloc(i32, ...)

declare void @_lfortran_printf(i8*, ...)
